sw/device/silicon_creator/lib/drivers/spi_device.h

```diff
@@ -74,11 +76,13 @@ enum {
   /**
    * Size of the payload area in spi_device buffer in bytes.
    */
-  kSpiDevicePayloadAreaNumBytes = 256,
+  kSpiDevicePayloadAreaNumBytes =
+      SPI_DEVICE_UPLOAD_STATUS2_PAYLOAD_START_IDX_MASK + 1,
```

Comment:
@alphan , this should partially address #11740 I think?

Reply:
Let's revert this change (we can also revert the one below) until #11740 is resolved properly. Then we can add `static_assert`s in the source file.

Reply:
Done